### ISSUE TITLE
Unable to Test With OpenCV File

### ISSUE THREAD

**Issue by vdaita** [ultralytics/yolov5#337]
Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following, otherwise it is non-actionable, and we can not help you:
 - **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
 - **Common dataset**: coco.yaml or coco128.yaml
 - **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#reproduce-our-environment
 
If this is a custom dataset/training question you **must include** your `train*.jpg`, `test*.jpg` and `results.png` figures, or we can not help you. You can generate these with `utils.plot_results()`.

## To Reproduce (REQUIRED)

Input:
```
import cv2
import torch
import numpy as np


from utils.datasets import *
from utils.utils import *

import base64



def apply1(img):

    # Padded resize
    img = letterbox(img, new_shape=416)[0]

    #cv2.imshow("hi", img)

    # Convert
    img = img[:, :, ::-1].transpose(2, 0, 1)  # BGR to RGB, to 3x416x416
    img = np.ascontiguousarray(img)


    device = torch_utils.select_device("cpu")
    half = False

    model = torch.load("mask.pt", map_location=device)['model'].float()
    model.to(device).eval()
    if half:
        model.half()  # to FP16

    names = model.module.names if hasattr(model, 'module') else model.names

    img = torch.from_numpy(img).to(device)
    img = img.half() if half else img.float()  # uint8 to fp16/32
    img /= 255.0  # 0 - 255 to 0.0 - 1.0
    if img.ndimension() == 3:
        img = img.unsqueeze(0)


    # Inference
    t1 = torch_utils.time_synchronized()
    pred = model(img, augment=False)[0]

    # Apply NMS
    pred = non_max_suppression(pred, 0.45, 5, classes=2, agnostic=True)
    t2 = torch_utils.time_synchronized()

    print(pred)

    faces = ""

    # Process detections
    for i, det in enumerate(pred):  # detections per image
        gn = torch.tensor(img.shape)[[1, 0, 1, 0]]  #  normalization gain whwh
        if det is not None and len(det):
            # Rescale boxes from img_size to im0 size
            det[:, :4] = scale_coords(img.shape[2:], det[:, :4], img.shape).round()

            # Print results
            for c in det[:, -1].unique():
                n = (det[:, -1] == c).sum()  # detections per class

            # Write results

            for *xyxy, conf, cls in det:
                if(names[int(cls)] == "nomask"):
                    im1 = img[int(xyxy[1]):int(xyxy[3]), int(xyxy[0]):int(xyxy[2])].copy()
                    _, im_arr = cv2.imencode('.jpg', im1)  # im_arr: image in Numpy one-dim array format.
                    im_bytes = im_arr.tobytes()
                    im_b64 = base64.b64encode(im_bytes)
                    faces += im_b64 + "\n"

    return faces


im1 = cv2.imread("/Users/vd/Files/yolov5/inference/images/bus.jpg")
# cv2.imshow("hi1", im1)
# cv2.waitKey(0)
print(apply1(im1))
```

Output:
```
Using CPU

[None]

```

## Expected behavior

I expected a list with the predictions. I have tested this image with the regular detect.py

## Environment

 - OS: [e.g. Ubuntu] macOS
 - GPU [e.g. 2080 Ti] None

**Comment by glenn-jocher** [comment]
@vdaita I'm sorry, we don't debug custom code. The default repo inference functionality is intact and correct, I suggest you start there.

I'll leave open for community feedback and remove the bug label.

**Comment by vdaita** [comment]
Alright, thank you!

**Comment by vdaita** [comment]
@glenn-jocher Does the repo have a script detect from cv2 image?

**Comment by glenn-jocher** [comment]
@vdaita yes, see the Google Colab notebook for examples.
`python detect.py --source image.jpg`
